test/lib/ipfs-set.js

```diff
@@ -0,0 +1,107 @@
+import ipfsAPI from 'ipfs-api';
+import IPFSSet from '../../lib/storage/IPFSSet.js';
+
+let ipfs = ipfsAPI('/ip4/127.0.0.1/tcp/5002');
+
+describe('IPFSSet', function () {
+  let ipfsSet;
+  let testData = ['test1', 'test2', 'test3', 'test4', 'test5'];
+
+  describe('Add to new set', () => {
+
+    before(async () => {
+      let ipfs = ipfsAPI('/ip4/127.0.0.1/tcp/5002');
+      ipfsSet = await new IPFSSet(ipfs).init();
+    });
+
+    it('add elements', async () => {
+      let hash1 = await ipfsSet.add(testData[0]);
+      let hash2 = await ipfsSet.add(testData[1]);
+      let links = ipfsSet.keys();
+      let data = await ipfsSet.values();
+
+      assert.equal(links.length, 2);
+      assert.equal(data.length, 2);
+      assert.deepEqual(data.sort(), [testData[0], testData[1]]);
+      assert.deepEqual(links.map((l) => l.multihash).sort(), [hash1, hash2].sort());
+    });
+  });
+
+  describe('Restore existing set', () => {
+    let hash1, hash2;
+    beforeEach(async () => {
+      let ipfs = ipfsAPI('/ip4/127.0.0.1/tcp/5002');
+      ipfsSet = await new IPFSSet(ipfs).init();
+      hash1 = await ipfsSet.add(testData[2]);
+      hash2 = await ipfsSet.add(testData[3]);
+    });
+
+    it('resotore from hash', async () => {
+      let restoredSet = await ipfsSet.initWithHash(ipfsSet.getOwnHash());
```

Comment:
IMO: you could move this to the beforeEach function, and the `restoredSet` variable into the describe scope;